### PR TITLE
[ION-938] Allow pointing our package.json to the slate-plugins fork

### PR DESCRIPTION
**Pull request by gitstart-firstresonance**
Credit cost: 120
## Checklist

- [x] The new code matches the existing patterns and styles.
- [x] The stories still work (run `yarn storybook`).
- [x] The stories are updated when relevant: `stories` for plugins, `knobs` for options.

### CODE REVIEW

**Comment by alantrrs** [comment]
@gitstart-firstresonance The objective here is just to allow us to point to this repo as a dependency from our package.json for when we're testing new things that haven't landed in the main repository. Let's not make any changes that are not required to make that happen. We want to continue to contribute to the main repository, not make this our own, so the fewer the changes, the better.

**Comment by gitstart-firstresonance** [comment]
> @gitstart-firstresonance The objective here is just to allow us to point to this repo as a dependency from our package.json for when we're testing new things that haven't landed in the main repository. Let's not make any changes that are not required to make that happen. We want to continue to contribute to the main repository, not make this our own, so the fewer the changes, the better.

Yes. The changes made were required to install this as a dependency as we would be building on installation. Other changes are github action workflow. This changes does not affect making contributing the changes back to the main repo

**Review by alantrrs — CHANGES_REQUESTED** [review]
- [x] Please provide an estimate for this. I don't see it here nor in the gitstart dashboard.
- [x] The default branch is `next`. Please branch out of there and make a PR against that branch
- [ ] Please only keep the specific changes that allow us to use it as dependency. I know other changes don't affect the build or contributing back, but I want this PR to be minimal.

**Comment by gitstart-firstresonance** [comment]
Unnecessary changes have been removed. Any other change is needed to enable seamless installation and development

**Review by alantrrs — DISMISSED** [review]
How are you testing that installing the repo works? After requiring it from tchami:
``` 
"@udecode/slate-plugins": "github:FirstResonance/slate-plugins#dab27a4e16f36c6030410cde98a85973db0abeb8",
```
I'm getting the following error:

image

**Comment by gitstart-firstresonance** [comment]
> How are you testing that installing the repo works? After requiring it from tchami:
> 
> ```
> "@udecode/slate-plugins": "github:FirstResonance/slate-plugins#dab27a4e16f36c6030410cde98a85973db0abeb8",
> ```
> 
> I'm getting the following error:
> 
> image

Dint't experince that which may be because the tchami branch used does not have the latest changes which uses `getSelectableElement` as it does not seem to be available in the slate-plugins next branch

**Comment by alantrrs** [comment]
@gitstart-firstresonance I see, seems like our `next` branch was outdated, I just pulled the latest from upstream. Please rebase the `next` branch.

**Comment by gitstart-firstresonance** [comment]
> @gitstart-firstresonance I see, seems like our `next` branch was outdated, I just pulled the latest from upstream. Please rebase the `next` branch.

ohh. Done

File: package.json [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
     "packages/core",
     "packages/slate-plugins"
   ],
+  "files": [
+    "packages",
+    "lerna.json"
+  ],
   "scripts": {
-    "build": "lerna run build --since",
+    "prepare": "yarn build",
```
Comment (alantrrs) [inline package.json:16]: Is this supposed to be using `yarn`?

**Review by alantrrs — CHANGES_REQUESTED** [review]
I'm still getting errors using it in tchami, now it seems there might be a react versions mismatch. I wonder if it's related to using 2 different package managers.
Can you look into it? Please verify by requiring it in tachmi.

image

**Review by alantrrs — APPROVED** [review]
This seems to work with tchami. What's the Gruntfile doing? - Could `npm dedupe` do the same job?

**Comment by gitstart-firstresonance** [comment]
> - This seems to work with tchami. What's the Gruntfile doing? - Could npm dedupe do the same job?

For some reason npm dedupe dint flatten the dependencies. The Gruntfile is basically removing any dependency that is declared as a peer dependency in the package.json

**Comment by alantrrs** [comment]
Ok. Closing this PR in favor of `fr-fork` branch which has the same content.

**Pull request closed without merging**